design/20200305-spec-v1alpha2-dependency-parampassing.md

```diff
@@ -0,0 +1,96 @@
+# Component and Trait Dependency with Parameter Passing
+
+* Author: @wonderflow @hongchaodeng
+
+## Background
+
+In https://github.com/oam-dev/spec/issues/171, we have discussed our dependency stories and started to explore solutions.
+Now that we have designed and implemented mechanism to satisfy dependency requirements, we are proposing to add them into OAM spec.
+
+## Use Cases
+
+### Case 1: MySQL and Web
+
+Two components, MySQL and Web exists in the same AppConfig and Web depends on MySQL that:
+
+1. MySQL needs to start first;
+2. MySQL connection info needs to be injected into Web's environment variables or files;
+
+### Case 2: NAS FileSystem and MountTarget
+
+Two components, NAS_FileSystem and NAS_MountTarget exists in the same AppConfig and NAS_MountTarget depends on NAS_FileSystem that:
+
+1. NAS_FileSystem needs to be created first;
+2. NAS_MountTarget needs filesystemID from NAS_FileSystem to fill in NAS_MountTarget's spec.
+
+## Proposal
```

Comment:
Can you explain a bit on why these two cases are different?

Reply:
Let me explain more.

The first case represents different OAM implementation cooperate with each other. In other word, system of Web component don't know what will output from system of Mysql component. 

The second case represents two components are in the same system, they can passthrough arguments with each other. One can output a key and finally fill in to the other one's spec.